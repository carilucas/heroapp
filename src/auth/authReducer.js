
import { LOGIN, LOGOUT } from './types';


export const authReducer = (state = {}, {type,payload}) => {
   switch (type) {
      case LOGIN:
        return payload
   
      case LOGOUT:
        return {logged:false}
   
      default:
         return state
   }
}
