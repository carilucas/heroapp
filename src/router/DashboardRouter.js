import React from 'react';
import {
   Route,
   Redirect,
   Switch
 } from "react-router-dom";
import { Navbar } from '../components/NavComponent';
import { DcComponent } from '../components/DcComponent';
import { MarvelComponent } from '../components/MarvelComponent';
import { SearchComponent } from '../components/SearchComponent';
import { HeroScreen } from '../components/HeroScreen';


export const DashboardRouter = () => {

   
   return (
      <div>
         <Navbar/>
         <div className="container mt-4">
               <Switch>
                  <Route exact path="/marvel" component={MarvelComponent}/>
                  <Route exact path="/dc" component={DcComponent}/>
                  <Route exact path="/search/" component={SearchComponent}/>
                  <Route exact path="/hero/:heroId" component={HeroScreen} />
                  <Redirect to='/marvel' />
               </Switch>
         </div>
      </div>
   )
}
