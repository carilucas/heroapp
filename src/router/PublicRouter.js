import React from 'react'
import { Redirect, Route } from 'react-router'
import { LoginComponent } from '../components/LoginComponent'

export const PublicRouter = ({
   logged,
   component:Component,
   ...rest
}) => {
   return (
      <Route 
         {...rest}
         component = {
            (props)=>(
               !logged
               
                  ? <LoginComponent {...props} /> 
                  : <Redirect to="/" />
            )
         }
      
      />
   )
}
