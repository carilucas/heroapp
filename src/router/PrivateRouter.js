import React from 'react'
import { Redirect, Route } from 'react-router'
import { DashboardRouter } from './DashboardRouter';
import PropTypes from 'prop-types'



export const PrivateRouter = ({
   logged,
   component:Component,
   ...rest
}) => {
   localStorage.setItem('lastPath',rest.location.pathname)
   return (
      <Route 
         {...rest}
         component = {
            (props)=>(
               logged
                  ? <DashboardRouter {...props} /> 
                  : <Redirect to="/login" />
            )
         }
      
      />
   )
}

PrivateRouter.propTypes = {
   logged : PropTypes.bool.isRequired,
   component : PropTypes.func.isRequired,
}