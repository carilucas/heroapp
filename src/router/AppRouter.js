import React, { useContext } from 'react'
import {
   BrowserRouter as Router,
   Switch
 } from "react-router-dom";
import { AuthCotext } from '../auth/authContext';
import { LoginComponent } from '../components/LoginComponent';
import { DashboardRouter } from './DashboardRouter';
import { PrivateRouter } from './PrivateRouter';
import { PublicRouter } from './PublicRouter';

export const AppRouter = () => {

  const {user} = useContext(AuthCotext);
  const {logged} = user;
  
   return (
      <Router>
      <div>
        <Switch>
            <PublicRouter exact path="/login" component={LoginComponent} logged={logged}/>
            <PrivateRouter path="/" component={DashboardRouter} logged={logged}/>
        </Switch>
      </div>
    </Router>
   )
}
