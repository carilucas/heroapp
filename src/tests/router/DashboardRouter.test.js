import { DashboardRouter } from "../../router/DashboardRouter";

import {mount} from 'enzyme';
import { AuthCotext } from "../../auth/authContext";
import { MemoryRouter } from "react-router";


describe('Pruebas en <DashboardRouter>', () => {
   
   const value = {
      user:{
         name:'Juan',
         logged: true,
      },
      dispatch: jest.fn()
   }

   test('Debe mostrar el snapshot', () => {
       
      const wrapper = mount(
         <MemoryRouter>
            <AuthCotext.Provider value={value}>
               <DashboardRouter/>
            </AuthCotext.Provider>
         </MemoryRouter>
      
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('h1').text().trim()).toBe('MarvelComponent');

   })
   
})
