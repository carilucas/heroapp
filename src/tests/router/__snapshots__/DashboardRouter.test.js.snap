// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRouter> Debe mostrar el snapshot 1`] = `"<div><nav class=\\"navbar navbar-expand-sm navbar-dark bg-dark\\"><div class=\\"container\\"><a class=\\"navbar-brand\\" href=\\"/\\">Hero App</a><div class=\\"navbar-collapse\\"><div class=\\"navbar-nav\\"><a class=\\"nav-item nav-link active\\" href=\\"/marvel\\" aria-current=\\"page\\">Marvel</a><a class=\\"nav-item nav-link\\" href=\\"/dc\\">DC</a><a class=\\"nav-item nav-link\\" href=\\"/search\\">Search</a></div></div><div class=\\"navbar-collapse collapse w-100 order-3 dual-collapse2\\"><ul class=\\"navbar-nav ms-auto\\"><li><span class=\\"nav-item nav-link\\" style=\\"color: cyan;\\">Juan</span></li><li><button class=\\"nav-item nav-link btn btn-outline-info\\">Logout</button></li></ul></div></div></nav><div class=\\"container mt-4\\"><div><h1>MarvelComponent</h1><hr><div class=\\"row animate__animated animate__fadeIn\\"><button class=\\"btn btn-primary\\">Sort</button><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-spider.jpg\\" class=\\"card-img-top\\" alt=\\"Spider Man\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Spider Man</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-captain.jpg\\" class=\\"card-img-top\\" alt=\\"Captain America\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Captain America</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-iron.jpg\\" class=\\"card-img-top\\" alt=\\"Iron Man\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Iron Man</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-thor.jpg\\" class=\\"card-img-top\\" alt=\\"Thor\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Thor</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-hulk.jpg\\" class=\\"card-img-top\\" alt=\\"Hulk\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Hulk</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-wolverine.jpg\\" class=\\"card-img-top\\" alt=\\"Wolverine\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Wolverine</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-daredevil.jpg\\" class=\\"card-img-top\\" alt=\\"Daredevil\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Daredevil</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-hawkeye.jpg\\" class=\\"card-img-top\\" alt=\\"Hawkeye\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Hawkeye</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-cyclops.jpg\\" class=\\"card-img-top\\" alt=\\"Cyclops\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Cyclops</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div><div class=\\"col-sm-4 my-3\\"><div class=\\"card\\"><img src=\\"./assets/heroes/marvel-silver.jpg\\" class=\\"card-img-top\\" alt=\\"Silver Surfer\\"><div class=\\"card-body\\"><h5 class=\\"card-title\\">Silver Surfer</h5><p class=\\"card-text\\">Marvel Comics</p><button class=\\"btn btn-primary\\">More Info</button></div></div></div></div></div></div></div>"`;
